plugin-server/src/main/ingestion-queues/session-recording-v2/sessions/recorder.ts

```diff
@@ -1,44 +1,43 @@
-import { Writable } from 'stream'
+import { Readable } from 'stream'
+import { createGzip } from 'zlib'
 
 import { ParsedMessageData } from '../kafka/types'
 
-interface WriteResult {
+export interface EndResult {
+    stream: Readable
     eventCount: number
-    bytesWritten: number
 }
 
 export class SessionRecorder {
-    private chunks: string[] = []
-    private size: number = 0
+    private readonly gzip: ReturnType<typeof createGzip>
```

Comment:
I feel like it would be useful if some of these classes had some comments describing what they do. 

For example for me it isn't clear at a glance if each line is gzip encoded or if it is the whole file 🤔 

If it is the whole file you probably want to have that on the Upload to be `ContentEncoding: 'gzip'` like the old consumer has